File: app/src/main/java/com/example/algorithm/HashTestB.java
package com.example.algorithm;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class HashTestB {
    public boolean solution(String[] phone_book) {
        boolean answer = true;
        for (int i = 0; i < phone_book.length - 1; i++){
            for (int l = i+1; l< phone_book.length; l++){
                if (phone_book[i].startsWith(phone_book[l])) {return false; }
                if (phone_book[l].startsWith(phone_book[i])) {return false; }
            }
        }
        return answer;
    }
}

File: app/src/main/java/com/example/algorithm/HashTestB.kt
package com.example.algorithm

class HashTestB {
    var result :Boolean = true
    fun solution(phonebook: Array<String>): Boolean{

        for (i in 0 .. phonebook.size -1){
            phonebook.forEach {
                if (phonebook[i] != it){
                    compare(phonebook[i], it)
                }
            }
        }

        return result
    }


    private fun compare(a : String, b : String){
        println("$a , $b")
        a.forEachIndexed { index, c ->

            if (index >= b.length || index >= a.length) return@forEachIndexed

            if (c == b[index]){
                println("$c , ${b[index]}")
                result = false
            }else{
                return
            }
        }

        return
    }
}